cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_provider.go

```diff
@@ -134,7 +135,9 @@ func BuildCloudProvider(name string, opts config.AutoscalingOptions, rl *cloudpr
 		return nil, fmt.Errorf("create cluster clientset failed: %v", err)
 	}
 
-	controller, err := newMachineController(kubeclient, clusterclient)
+	enableMachineDeployments := os.Getenv("OPENSHIFT_MACHINE_API_CLOUDPROVIDER_ENABLE_MACHINE_DEPLOYMENTS") != ""
```

Comment:
`OPENSHIFT_MACHINE_API_CLOUDPROVIDER_ENABLE_MACHINE_DEPLOYMENTS` is so long noone will want to type it voluntary. Though, I am still for adding `--feature-gates` option as is in [1] at some point.

[1] https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/

Reply:
Open to suggestions for the environment name. Looking at the cluster-autoscaler-operator I see:

```yaml
        env:
        - name: WATCH_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: CLUSTER_AUTOSCALER_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: LEADER_ELECTION_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: CLUSTER_AUTOSCALER_IMAGE
          value: docker.io/openshift/origin-cluster-autoscaler:v4.0
```

How about `CLUSTER_AUTOSCALER_ENABLE_MACHINE_DEPLOYMENTS`?

CC @bison